src/components/api/RoiBox.tsx

```diff
@@ -7,18 +7,30 @@ import { Box } from '../Box';
 import { RoiBoxCorner } from '../RoiBoxCorner';
 
 import { RoiListProps } from './RoiList';
+import { useRoiDispatch } from '../../hooks/useRoiDispatch';
 
 interface RoiBoxProps {
   roi: Roi;
   getStyle: RoiListProps['getStyle'];
+  getReadOnly: RoiListProps['getReadOnly'];
 }
 
 function RoiBoxInternal(props: RoiBoxProps): JSX.Element {
-  const { roi, getStyle } = props;
+  const { roi, getStyle, getReadOnly } = props;
   const roiState = useRoiState();
 
   const { x, y, width, height, id, label } = roi;
-  const isActive = id === roiState.selectedRoi;
+  const isSelected = id === roiState.selectedRoi;
+  const readOnly = getReadOnly(roi);
+
+  const roiDispatch = useRoiDispatch();
+
+  useEffect(() => {
+    if (readOnly) {
+      roiDispatch({ type: 'CANCEL_ACTION' });
+      roiDispatch({ type: 'UNSELECT_ROI', payload: id });
```

Comment:
I think here it's better to only send the unselect dispatch and handle the cancelation inside the reducer when the payload matches the selected roi. It will avoid unnecessarily cancelling an ongoing action.